src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -1294,6 +1294,14 @@ final public Object invokeOn(boolean blockLocally, Object obj, String methodName
       }
       retobj = method.invoke(obj, params);
       if (blockLocally) {
+        Outbox outbox = null;
+        if (obj instanceof ServiceInterface) {
+          outbox = ((ServiceInterface)obj).getOutbox();
```

Comment:
What does this do?

Reply:
gets the target si's outbox .. invokeOn isn't necessarily the service that is invoking itself, it may be invoking on another service, so its the outbox of the "invoked" service that is required here ...
most of the time it is the same, but not guaranteed .. its a bug

Reply:
It doesn't seem to be used anywhere though, the outbox declared in this if statement is not accessible outside of it so it's not shadowing `this.outbox` later

Reply:
broadcast can invokeOn .. when called it invokeOn inside of itself on a different service .. so when it tries to do this and uses its own outbox, you won't get the correct outbox that needs to be blocking ..

Reply:
I understand that, but this code doesn't seem to do anything with the foreign outbox, it just gets it and then the if statement ends. Since Java has block scoping, the newly assigned outbox variable is not accessible outside of the if statement, so effectively this conditional doesn't do anything

Reply:
![image](https://github.com/MyRobotLab/myrobotlab/assets/1266144/86ee9afa-18bb-42c0-b518-495de81b7ff5)

Reply:
1 - beginning of scope
2 - assignment
3 - usage

Reply:
if the target is another local service then it needs that services notficationList to process the other subscriptions

Reply:
Oh, I misread the code, I thought the outer if statement was closed before the usage. My mistake